iroh-rpc-types/proto/p2p.proto

```diff
@@ -95,28 +97,48 @@ message GetPeersResponse {
   map<string, Multiaddrs> peers = 1;
 }
 
-message ConnectRequest {
-  // Serialized peer id
-  bytes peer_id = 1; 
-  // Serialized list of multiaddrs
-  repeated bytes addrs = 2;
+message ConnectByPeerIdRequest {
+  bytes peer_id = 1;
 }
 
-message ConnectResponse {
-  bool success = 1; 
+message ConnectRequest {
+  // Serialized PeerId
+  bytes peer_id = 1;
+  // Serialized multiaddr
+  repeated bytes addrs = 2;
 }
 
 message DisconnectRequest {
   // Serialized peer id
   bytes peer_id = 1;
 }
 
+message LookupRequest {
+  // PeerId
+  bytes peer_id = 1;
+  // Serialized multiaddr
+  optional bytes addr = 2;
+}
+
+message PeerInfo {
+  // PublicKey
+  bytes peer_id = 1;
```

Comment:
Somehow it is sad that we have this awesome type system in rust, yet we drop to the level of blobs whenever we do RPC...

Reply:
Not only sad for performance but annoying af while writing it!!!

Reply:
it will be gone sooooon :)